integrations/clickup/main.py

```diff
@@ -0,0 +1,170 @@
+            for webhook in webhooks:
+                if webhook["endpoint"] == f"{app_host}/integration/webhook":
+                    logger.info(f"Webhook already exists for team {team['id']}")
+                    WEBHOOK_TEAM_MAP[webhook["id"]] = team["id"]
+                    break
+                else:
+                    logger.info(f"Creating webhook for team {team['id']}")
+                    webhook = await clickup_client.create_clickup_webhook(
+                        team["id"], app_host
+                    )
+                    WEBHOOK_TEAM_MAP[webhook["id"]] = team["id"]
+    logger.warning(f"Webhook team map: {WEBHOOK_TEAM_MAP}")
+
+
+@ocean.on_resync(ObjectKind.TEAM)
+async def on_resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for teams in clickup_client.get_clickup_teams():
+        logger.info(f"Received batch of {len(teams)} teams")
+        yield teams
+
+
+@ocean.on_resync(ObjectKind.SPACE)
+async def on_resync_spaces(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for spaces in clickup_client.get_all_spaces():
+        logger.info(f"Received batch of {len(spaces)} spaces")
+        yield spaces
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for projects in clickup_client.get_all_projects():
+        logger.info(f"Received batch of  {len(projects)} projects")
+        yield projects
+
+
+@ocean.on_resync(ObjectKind.ISSUE)
+async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for issues in clickup_client.get_paginated_issues():
+        logger.info(f"Received batch of {len(issues)} issues")
+        yield issues
+
+
+async def handle_register(
+    clickup_client: Any,
+    entity_id: str,
+    kind: ObjectKind,
+    event_type: str,
+    webhook_id: str,
+) -> None:
+    team_id = WEBHOOK_TEAM_MAP.get(webhook_id)
+    if kind == ObjectKind.ISSUE:
+        entity = await clickup_client.get_single_issue(entity_id)
+    elif kind == ObjectKind.PROJECT:
+        entity = await clickup_client.get_single_project(entity_id, team_id)
+    else:
+        entity = await clickup_client.get_single_space(entity_id, team_id)
+
+    if entity:
+        await ocean.register_raw(kind, [entity])
+        logger.info(f"Registered {kind} for event {event_type}")
+    else:
+        logger.error(f"Handler returned None for entity_id {entity_id}")
+
+
+async def handle_unregister(entity_id: str, kind: ObjectKind, event_type: str) -> None:
+    try:
+        await ocean.unregister_raw(kind, [{"id": entity_id}])
+        logger.info(f"Unregistered {kind} for event {event_type}")
+    except Exception as e:
+        logger.error(f"Exception {e} occurred while attempting to unregister raw")
+
```

Comment:
I feel like this logic becomes quite big, I would suggest assigning it a separate file which will contain the logic of handling webhooks, this will make the code more readable

Reply:
It has been seperated